### PR TITLE
Add support for leveraging namePrefix in Java SDK

### PR DESCRIPTION
**Pull request by checketts**
Also warn if trying to check for toggles that don't match the prefix to avoid confusion for why it is always disabled.

### CODE REVIEW

File: src/test/java/no/finn/unleash/event/SubscriberTest.java [hunk 1]
```diff
@@ -61,15 +65,14 @@ void subscriberAreNotified() {
...
-        assertThat(testSubscriber.events)
-                .hasSize(
-                        1 // Bootstrapping
-                                + 3 // feature evaluations
-                                + 2 // toggle fetches
-                                + 1 // unleash ready
-                                + 1 // client registration
-                                + 1 // client metrics
-                        );
+//        assertThat(testSubscriber.events).filteredOn(e -> e instanceof ToggleBootstrapHandler.ToggleBootstrapRead).hasSize(1);
```
Comment (checketts) [inline src/test/java/no/finn/unleash/event/SubscriberTest.java:68]: I had to update this test to get it passing (it was failing in master already)

File: src/main/java/no/finn/unleash/repository/ToggleBootstrapHandler.java [hunk 2]
```diff
@@ -43,7 +43,7 @@ public ToggleCollection read() {
         return new ToggleCollection(Collections.emptyList());
     }
 
-    private static class ToggleBootstrapRead implements UnleashEvent {
+    public static class ToggleBootstrapRead implements UnleashEvent {
```
Comment (checketts) [inline src/main/java/no/finn/unleash/repository/ToggleBootstrapHandler.java:46]: I had to make this public so the event could be introspected in the test (I can't imagine we want the event to not be visible to subscribers.)

**Comment by ivarconr** [comment]
Read quickly through, and the only concern i have is if the implementation will allow multiple query fields at the same time (e.g. project and namsPrefix)?

**Comment by checketts** [comment]
This implementation does not support that. I was just reading through the node code and realized that was supported, so I'll update this PR to support that as well.

**Comment by checketts** [comment]
I've updated the code to support both project and namePrefix now.

**Comment by checketts** [comment]
I just added the changelog entry. Sorry I had forgotten that before!

File: src/main/java/no/finn/unleash/util/UnleashURLs.java [hunk 3]
```diff
@@ -25,22 +28,35 @@ public URL getFetchTogglesURL() {
...
+        appendParam(suffix, "namePrefix", namePrefix);
+
+        try {
+            return URI.create(fetchTogglesURL + suffix.toString()).normalize().toURL();
+        } catch (IllegalArgumentException|MalformedURLException e) {
+            throw new IllegalArgumentException("fetchTogglesURL [" + fetchTogglesURL + suffix + "] was not URL friendly.", e);
+        }
+    }
+
+    private void appendParam(StringBuilder suffix, String key, @Nullable String value) {
```
Comment (ivarconr) [inline src/main/java/no/finn/unleash/util/UnleashURLs.java:51]: @chriswk will this also be usable for adding array of tags later on?

**Review by ivarconr — APPROVED** [review]
Thanks for yet another great contribution!

LGTM!

**Pull request merged**